bazel/BUILD

```diff
@@ -0,0 +1,13 @@
+load("@bazel_skylib//rules:common_settings.bzl", "bool_flag")
+
+bool_flag(
+    name = "mark_optimized",
+    build_setting_default = False,
+)
+
+config_setting(
+    name = "optimized_build",
+    flag_values = {
+        ":mark_optimized": "true",
+    },
+)
```

Comment:
This feels a little heavy handed. I mainly don't like that i has to be manually set when any optimization flags are passed in.

However, i couldn't find any elegant way to detect optimization flags. If you know of a nicer way, do let me know.

Reply:
Otherwise you can check for `--compliation_mode` via:

```
config_setting(
    name = "opt_build",
    values = {
        "compilation_mode": "opt",
    },
)
```

Reply:
it's also the sanitizer that add `-O1`. This also fails.

Reply:
> I think I would rather explicitly opt into running the fuzz tests

this makes sense. The only issue is that it is not clear when they should/could be run.
Should we just leave a comment on the fuzz rule saying that it only works with unoptimized builds?

Reply:
Which is why I'd rather have a special `--config=fuzz` that enables the exact set of compilation flags required for fuzz testing, then we have some global flag that all fuzz tests can check for - similar to what you have today but instead of `mark_optimized` just do `fuzzing_enabled` or something and we have a .bazelrc configuration for making sure all the build flags work out.

I'm also confused why `-O1` fails because our current CMake debug config uses `-O1` IIRC